src/manifolds/EmbeddedManifold.jl

```diff
@@ -0,0 +1,82 @@
+"""
+    AbstractEmbeddedManifold
+
+"""
+abstract type AbstractEmbeddedManifold <: AbstractDecoratorManifold end
+
+abstract type AbstractEmbeddingType end
+
+"""
+    IsometricEmbedding <: AbstractEmbeddingType
+
+"""
+struct IsometricEmbedding <: AbstractEmbeddingType end
+
+struct EmbeddedManifold{MT <: Manifold, NT <: Manifold, ET <: AbstractEmbeddingType} <: AbstractEmbeddedManifold
```

Comment:
I'm not sure it's specific enough to be a struct. I'd expect objects of `AbstractEmbeddedManifold` to represent uniquely defined embeddings (and `EmbeddedManifold` doesn't have enough information in it to do that).

Reply:
What are you missing in order to be specific enough?
Two manifolds (`Sphere(2)` and `Euclidean(3)` for example) should should make it specific enough. The `EmbeddingType` is already a property I want to dispatch on later.

And the `AbstractEmbeddedManifold` is meant for example for `SymmetricMatrices` where I have something special in mind, namely that the symmetric matrices just hold their embedding to dispatch inner and so on onto but are themselves also the manifold within the embedding.

Reply:
I can embed a sphere in R^3 in different ways, for example the center can be at any point. Is this structure supposed to represent singular canonical embeddings of a single manifold in other manifolds? Could you give an example for that?

Reply:
the different ways can be done with the `EmbeddingType`. It is meant to represent _any_ embedding of a manifold `M` in an embedding `N`. Where the `EmbeddingType` might imply certain properties – for example for the `IsometricEmbedding` the `inner` from the embedding will be used.

Reply:
Examples will follow when I am done with the decorator rework. I already have a few in mind (`Sphere`will reduce in code, `SymmetricMatrices` nearly have no extra code necessary`, `SkewSymmetricMatrices`will act similar as `SymmetricMatrices`, maybe even a `QuotientManifold`.

Reply:
> the different ways can be done with the `EmbeddingType`

OK, then `IsometricEmbedding` should be an abstract type (and then we may have something like a concrete type `DefaultIsometricEmbedding`). It should also be documented how tangent space embedding is related to these embedding types. I wonder if representing tangent vectors on Lie groups as elements of the corresponding Lie algebra could cause some trouble here.

Reply:
Yes, of course we can do an `AbstractIsometricEmbedding`.

For the effect on representing tangents on the Lie algebra, I am not sure, I was thinking, just passing that through to the group, this decorator should not cause any effect.